doc/ROADMAP.rst.txt

```diff
@@ -0,0 +1,333 @@
+including #1323, #1903, #2045 and #2640.
+
+The morphology interface needs to be standardized:
+
+  - binary dilation/erosion/opening/closing take a "structure" argument,
+    whereas their grey equivalent take size (has to be a tuple, not a scalar),
+    footprint, or structure.
+  - a scalar should be acceptable for size, equivalent to providing that same
+    value for each axis.
+  - for binary dilation/erosion/opening/closing, the structuring element is
+    optional, whereas it's mandatory for grey.  Grey morphology operations
+    should get the same default.
+  - other filters should also take that default value where possible.
+
+
+odr
+---
+Rename the module to ``regression`` or ``fitting``, include
+``optimize.curve_fit``. This module will then provide a home for other fitting
+functionality - what exactly needs to be worked out in more detail, a
+discussion can be found at https://github.com/scipy/scipy/pull/448.
+
+
+optimize
+````````
+Overall this module is in reasonably good shape, however it is missing a few
+more good global optimizers as well as large-scale optimizers.  These should be
+added.  Other things that are needed:
+
+  - deprecate the ``fmin_*`` functions in the documentation, ``minimize`` is
+    preferred.
+  - clearly define what's out of scope for this module.
+
+
+signal
+``````
+*Convolution and correlation*: (Relevant functions are convolve, correlate,
+fftconvolve, convolve2d, correlate2d, and sepfir2d.) Eliminate the overlap with
+`ndimage` (and elsewhere).  From `numpy`, `scipy.signal` and `scipy.ndimage`
+(and anywhere else we find them), pick the "best of class" for 1-D, 2-D and n-d
+convolution and correlation, put the implementation somewhere, and use that
+consistently throughout scipy.
+
+*B-splines*: (Relevant functions are bspline, cubic, quadratic, gauss_spline,
+cspline1d, qspline1d, cspline2d, qspline2d, cspline1d_eval, and spline_filter.)
+Move the good stuff to `interpolate` (with appropriate API changes to match how
+things are done in `interpolate`), and eliminate any duplication.
+
+*Filter design*: merge `firwin` and `firwin2` so `firwin2` can be removed.
+
+*Continuous-Time Linear Systems*: remove `lsim2`, `impulse2`, `step2`.  Make
+`lsim`, `impulse` and `step` "just work" for any input system.  Improve
+performance of ltisys (less internal transformations between different
+representations).
+
+*Wavelets*: add proper wavelets, including discrete wavelet transform.  What's
+there now doesn't make much sense.
+
+
+sparse
+``````
+The sparse matrix formats are getting feature-complete but are slow ...
+reimplement parts in Cython?
+
+    - Small matrices are slower than PySparse, needs fixing
+
+There are a lot of formats.  These should be kept, but
+improvements/optimizations should go into CSR/CSC, which are the preferred
+formats.  LIL may be the exception, it's inherently inefficient.  It could be
+dropped if DOK is extended to support all the operations LIL currently
+provides.
```

Comment:
Been a while :-). I played a bit with wrapping `std::map`, and it seems reasonable so far. Comments welcome: https://github.com/ev-br/sparr
(this is outside of scipy for now just for convenience, can be moved to the scipy umberlla if desired)